### PR TITLE
Sprint task 1063 create manuscript table and model class

### PR DESCRIPTION
**Pull request by kencho51**
# Pull request for issue: #1063 

This is a pull request for the following functionalities:

To create manuscript table using migration script and manuscript model class using yii2 gii tool.

## How to test and the expected results?

```
% cd gigareview
% ./up.sh
% docker-compose run --rm console ./vendor/codeception/codeception/codecept -c common/codeception.yml run -v unit ManuscriptTest
Creating review_console_run ... done
Codeception PHP Testing Framework v4.2.0 https://helpukrainewin.org
Powered by PHPUnit 8.5.27 #StandWithUkraine

Common\tests.unit Tests (2) --------------------------------------------------------------------------------------------------------------------
Modules: Yii2
------------------------------------------------------------------------------------------------------------------------------------------------
✔ ManuscriptTest: Can create timestamp (1.13s)
✔ ManuscriptTest: Can insert values to table (0.07s)
------------------------------------------------------------------------------------------------------------------------------------------------


Time: 1.58 seconds, Memory: 12.00 MB

OK (2 tests, 12 assertions)

```

## How have functionalities been implemented?

Describe how the new functionalities have been implemented by the changed code at a high level

The migration script and the model class were created yii cmd tool:
```
# To create migration script
% % docker-compose run --rm console ./yii migrate/create create_manuscript_table --fields="doi:integer,manuscript_number:string,article_title:string,publication_date:date,editorial_status:string,editorial_status_date:date,created_at:biginteger,updated_at:biginteger" 
# To create model class
% docker-compose run --rm console ./yii gii/model --ns="common\models" --tableName="manuscript" --modelClass="Manuscript"
# To create unit test
% docker-compose run --rm console ./vendor/codeception/codeception/codecept -c /app/common/codeception.yml generate:test unit Manuscript

```

## Any issues with implementation?

None

## Any changes to automated tests?

None

## Any changes to documentation?

None

## Any technical debt repayment?

None

## Any improvements to CI/CD pipeline?

None

### CODE REVIEW

File: gigareview/console/migrations/m220629_145450_create_manuscript_table.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+     */
+    public function safeUp()
+    {
+        $this->createTable('{{%manuscript}}', [
+            'id' => $this->primaryKey(),
+            'manuscript_number' => $this->string(),
+            'article_title' => $this->string(),
+            'revision_number' => $this->smallinteger(),
+            'created_at' => $this->bigInteger(),
+            'updated_at' => $this->bigInteger(),
```
Comment (pli888) [inline gigareview/console/migrations/m220629_145450_create_manuscript_table.php]: * `revision_number` is not required, and neither is `revison_id`
* Keep `created_at` and `updated_at` for table timestamps
* Need a `publication_date` field
* Need a `doi` field
* Need an `editorial_status_date` field
* Need an `editorial_status` field
* Update ER diagram using omnigraffle

File: gigareview/common/tests/unit/ManuscriptTest.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        $this->assertNull($manuscript->created_at);
+        $this->assertNull($manuscript->updated_at);
+        $manuscript->save();
+        $this->assertNotNull($manuscript->created_at);
+        $manuscript->manuscript_number = "Test-GIGA-D-22-12345";
+        sleep(1);
+        $manuscript->save();
+        $this->assertGreaterThan($manuscript->created_at, $manuscript->updated_at);
+    }
+    public function testCanInsertValuesToTable()
```
Comment (pli888) [inline gigareview/common/tests/unit/ManuscriptTest.php]: This function is not required.

File: gigareview/common/models/Manuscript.php [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+ * This is the model class for table "manuscript".
+ *
+ * @property int $id
+ * @property string|null $manuscript_number
+ * @property string|null $article_title
+ * @property int|null $revision_number
+ * @property int|null $created_at
+ * @property int|null $updated_at
+ */
+class Manuscript extends \yii\db\ActiveRecord
```
Comment (pli888) [inline gigareview/common/models/Manuscript.php:21]: This function needs to be updated according to changes required in `m220629_145450_create_manuscript_table`.

**Review by pli888 — CHANGES_REQUESTED** [review]
See comments in code.

**Review by rija — APPROVED** [review]
Looks good to me @kencho51

**Review by pli888 — APPROVED** [review]
The code looks good now.

**Pull request merged**